trl/commands/sft.py

```diff
@@ -0,0 +1,148 @@
+from tqdm.rich import tqdm
+from transformers import AutoTokenizer, HfArgumentParser, TrainingArguments
+
+from trl import ModelConfig, RichProgressCallback, SFTTrainer, get_peft_config, get_quantization_config
+
+
+tqdm.pandas()
+
+logging.basicConfig(format=FORMAT, datefmt="[%X]", handlers=[RichHandler()], level=logging.INFO)
+
+
+@dataclass
+class ScriptArguments:
+    dataset_name: str = field(default="timdettmers/openassistant-guanaco", metadata={"help": "the dataset name"})
+    dataset_text_field: str = field(default="text", metadata={"help": "the text field of the dataset"})
+    max_seq_length: int = field(default=512, metadata={"help": "The maximum sequence length for SFT Trainer"})
+
+
+if __name__ == "__main__":
+    console = Console()
+    parser = HfArgumentParser((ScriptArguments, TrainingArguments, ModelConfig))
+    args, training_args, model_config = parser.parse_args_into_dataclasses()
+    training_args.gradient_checkpointing_kwargs = dict(use_reentrant=False)
+
+    # Force use our print callback
+    training_args.disable_tqdm = True
+
+    ################
+    # Model & Tokenizer
+    ################
+    torch_dtype = (
+        model_config.torch_dtype
+        if model_config.torch_dtype in ["auto", None]
+        else getattr(torch, model_config.torch_dtype)
+    )
+    quantization_config = get_quantization_config(model_config)
+    model_kwargs = dict(
+        revision=model_config.model_revision,
+        trust_remote_code=model_config.trust_remote_code,
+        attn_implementation=model_config.attn_implementation,
+        torch_dtype=torch_dtype,
+        use_cache=False if training_args.gradient_checkpointing else True,
+        # device_map=get_kbit_device_map() if quantization_config is not None else None,
+        quantization_config=quantization_config,
+    )
+    tokenizer = AutoTokenizer.from_pretrained(model_config.model_name_or_path, use_fast=True)
+    tokenizer.pad_token = tokenizer.eos_token
+
+    ################
+    # Dataset
+    ################
+    raw_datasets = load_dataset(args.dataset_name)
+    train_dataset = raw_datasets["train"]
+    eval_dataset = raw_datasets["test"]
+
+    ################
+    # Training
+    ################
+    with console.status("[bold green]Initializing the SFTTrainer..."):
+        trainer = SFTTrainer(
+            model=model_config.model_name_or_path,
+            model_init_kwargs=model_kwargs,
+            args=training_args,
+            train_dataset=train_dataset,
+            eval_dataset=eval_dataset,
+            dataset_text_field="text",
+            max_seq_length=args.max_seq_length,
+            tokenizer=tokenizer,
+            packing=True,
+            peft_config=get_peft_config(model_config),
+            callbacks=[RichProgressCallback],
+        )
+
+    trainer.train()
+
+    with console.status("[bold green]Training completed ! saving the model ..."):
+        trainer.save_model(training_args.output_dir)
```

Comment:
```suggestion
    with console.status(f"[bold green]Training completed! Saving the model to {training_args.output_dir}"):
        trainer.save_model(training_args.output_dir)
```

and we should also push to hub by default :)